File: android-templates/testData/golden/testNewNativeCppActivity/Template test module/src/main/java/template/test/pkg/MainActivity.java
package template.test.pkg;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import template.test.pkg.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'pkg' library on application startup.
    static {
        System.loadLibrary("pkg");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
    }

    /**
     * A native method that is implemented by the 'pkg' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: android-templates/testData/golden/testNewNativeCppActivityWithKotlin/Template test module/src/main/java/template/test/pkg/MainActivity.kt
package template.test.pkg

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import template.test.pkg.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Example of a call to a native method
        binding.sampleText.text = stringFromJNI()
    }

    /**
     * A native method that is implemented by the 'pkg' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    companion object {
        // Used to load the 'pkg' library on application startup.
        init {
            System.loadLibrary("pkg")
        }
    }
}